bundler/exe/bundle

```diff
@@ -18,14 +18,16 @@ end
 # Workaround for non-activated bundler spec due to missing https://github.com/rubygems/rubygems/commit/4e306d7bcdee924b8d80ca9db6125aa59ee4e5a3
 gem "bundler", Bundler::VERSION if Gem.rubygems_version < Gem::Version.new("2.6.2")
 
-# Check if an older version of bundler is installed
-$LOAD_PATH.each do |path|
-  next unless path =~ %r{/bundler-0\.(\d+)} && $1.to_i < 9
-  err = String.new
-  err << "Looks like you have a version of bundler that's older than 0.9.\n"
-  err << "Please remove your old versions.\n"
-  err << "An easy way to do this is by running `gem cleanup bundler`."
-  abort(err)
+if Gem::VERSION < "3.2.3" && RUBY_VERSION < "2.6.a" && !ENV["BUNDLER_NO_OLD_RUBYGEMS_WARNING"]
+  Bundler.ui.warn \
+    "Your rubygems version (#{Gem::VERSION})) has a bug that makes dropping " \
+    "support for old rubies in bundler through bumping the " \
+    "`required_ruby_version` field in the bundler gemspec unsafe for any " \
+    "scripts including `gem install bundler`. That means that, if those " \
+    "scripts are using rubygems #{Gem::VERSION}, they will break as soon " \
+    "as bundler releases a version dropping support for your ruby version. " \
+    "Please upgrade rubygems by running `gem update --system` to avoid future " \
+    "breakage and silence this warning."
```

Comment:
```suggestion
    "breakage and silence this warning, or use `gem install bundler:'< 2.3.0'`."
```
The new message is much better to explain the issue, thanks :)

Reply:
I was late to the party but it looks like this is a reasonable message and spells out the two main options.  I agree with @eregon that if you are explicitly pinned to an older version of Ruby (whether from JRuby still supporting it or other reasons) you would find the warning annoying.  I also agree it is nice to give a heads up that something will break at some future point.

Reply:
The recommendation of downgrading bundler requires further actions to work, so it will need further clarifications. And I'm not a fan of recommending downgrading the software that I help maintain anyways. Educated users can do this if they please even if we don't directly recommended, so I think it's fine as is.